### PR TITLE
[REVIEW] Fix `loc` and `iloc` doc 

### LINKED ISSUE #5474: [DOC] iloc discrepancy between cudf and pandas
**Describe the bug**
The iloc function in Pandas returns a pandas.Series object.  Sometimes the iloc function in Cudf returns a cudf.DataFrame.  It seems to be the case when the types of the columns do not match.  This makes code that expects a series fail.

**Steps/Code to reproduce bug**
Python fragment:
```
a = pd.DataFrame({'x':['1', '2'], 'y':[1, 2]})
b = cudf.DataFrame({'x':['1', '2'], 'y':[1, 2]})

print(a.iloc[1])
print(type(a.iloc[1]))
print(b.iloc[1])
print(type(b.iloc[1]))

print('a: ', a.iloc[1][0])
print('b: ', b.iloc[1][0])
```

Output is:
```
x    2
y    2
Name: 1, dtype: object
<class 'pandas.core.series.Series'>
   x  y
1  2  2
<class 'cudf.core.dataframe.DataFrame'>
a:  2
```
followed by a stack trace:
```
KeyError                                  Traceback (most recent call last)
<ipython-input-27-582a65142a8a> in <module>
     14 
     15 print('a: ', a.iloc[1][0])
---> 16 print('b: ', b.iloc[1][0])

~/anaconda3/envs/cugraph_dev/lib/python3.7/contextlib.py in inner(*args, **kwds)
     72         def inner(*args, **kwds):
     73             with self._recreate_cm():
---> 74                 return func(*args, **kwds)
     75         return inner
     76 

~/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cudf/core/dataframe.py in __getitem__(self, arg)
    506         """
    507         if is_scalar(arg) or isinstance(arg, tuple):
--> 508             return self._get_columns_by_label(arg, downcast=True)
    509 
    510         elif isinstance(arg, slice):

~/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cudf/core/frame.py in _get_columns_by_label(self, labels, downcast)
    226         If downcast is True, try and downcast from a DataFrame to a Series
    227         """
--> 228         new_data = self._data.get_by_label(labels)
    229         if downcast:
    230             if is_scalar(labels):

~/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cudf/core/column_accessor.py in get_by_label(self, key)
    215                 if any(isinstance(k, slice) for k in key):
    216                     return self.get_by_label_with_wildcard(key)
--> 217             return self.get_by_label_grouped(key)
    218 
    219     def get_by_label_list_like(self, key):

~/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cudf/core/column_accessor.py in get_by_label_grouped(self, key)
    225 
    226     def get_by_label_grouped(self, key):
--> 227         result = self._grouped_data[key]
    228         if isinstance(result, cudf.core.column.ColumnBase):
    229             return self.__class__({key: result})

KeyError: 0
```

**Expected behavior**
Should generate same behavior as Pandas

**Environment overview (please complete the following information)**
 - Environment location: Bare Metal
 - Method of cuDF install: conda

**Environment details**
<details><summary>Click here to see environment details</summary><pre>
     
     **git***
     commit 8ee2106e7a2b07a84384e87deb8ec5a724a8c7ae (HEAD -> branch-0.15, origin/branch-0.15)
     Merge: 91d54e5 d5fbbf8
     Author: gpuCI <38199262+GPUtester@users.noreply.github.com>
     Date:   Tue May 26 10:44:23 2020 -0400
     
     Merge pull request #901 from rapidsai/branch-0.14
     
     [gpuCI] Auto-merge branch-0.14 to branch-0.15 [skip ci]
     **git submodules***
     
     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=16.04
     DISTRIB_CODENAME=xenial
     DISTRIB_DESCRIPTION="Ubuntu 16.04.5 LTS"
     NAME="Ubuntu"
     VERSION="16.04.5 LTS (Xenial Xerus)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 16.04.5 LTS"
     VERSION_ID="16.04"
     HOME_URL="http://www.ubuntu.com/"
     SUPPORT_URL="http://help.ubuntu.com/"
     BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
     VERSION_CODENAME=xenial
     UBUNTU_CODENAME=xenial
     Linux charlesh-linux 4.15.0-76-generic #86~16.04.1-Ubuntu SMP Mon Jan 20 11:02:50 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
     
     ***GPU Information***
     Mon Jun 15 18:47:49 2020
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 418.87.00    Driver Version: 418.87.00    CUDA Version: 10.1     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |===============================+======================+======================|
     |   0  Quadro GV100        Off  | 00000000:15:00.0 Off |                  Off |
     | 36%   49C    P2    42W / 250W |   1000MiB / 32476MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   1  Quadro GV100        Off  | 00000000:2D:00.0 Off |                  Off |
     | 41%   54C    P2    39W / 250W |     11MiB / 32478MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     
     +-----------------------------------------------------------------------------+
     | Processes:                                                       GPU Memory |
     |  GPU       PID   Type   Process name                             Usage      |
     |=============================================================================|
     |    0     27023      C   ...h/anaconda3/envs/cugraph_dev/bin/python   989MiB |
     +-----------------------------------------------------------------------------+
     
     ***CPU***
     Architecture:          x86_64
     CPU op-mode(s):        32-bit, 64-bit
     Byte Order:            Little Endian
     CPU(s):                12
     On-line CPU(s) list:   0-11
     Thread(s) per core:    2
     Core(s) per socket:    6
     Socket(s):             1
     NUMA node(s):          1
     Vendor ID:             GenuineIntel
     CPU family:            6
     Model:                 85
     Model name:            Intel(R) Xeon(R) Gold 6128 CPU @ 3.40GHz
     Stepping:              4
     CPU MHz:               1200.353
     CPU max MHz:           3700.0000
     CPU min MHz:           1200.0000
     BogoMIPS:              6800.00
     Virtualization:        VT-x
     L1d cache:             32K
     L1i cache:             32K
     L2 cache:              1024K
     L3 cache:              19712K
     NUMA node0 CPU(s):     0-11
     Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd mba ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req pku ospke md_clear flush_l1d
     
     ***CMake***
     /usr/bin/cmake
     cmake version 3.5.1
     
     CMake suite maintained and supported by Kitware (kitware.com/cmake).
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
     Copyright (C) 2015 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     /usr/local/cuda/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2019 NVIDIA Corporation
     Built on Sun_Jul_28_19:07:16_PDT_2019
     Cuda compilation tools, release 10.1, V10.1.243
     
     ***Python***
     /home/charlesh/anaconda3/bin/python
     Python 3.7.4
     
     ***Environment Variables***
     PATH                            : /home/charlesh/bin:/home/charlesh/.local/bin:/home/charlesh/anaconda3/bin:/home/charlesh/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/local/cuda/bin
     LD_LIBRARY_PATH                 : :/usr/local/cuda/lib64
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /home/charlesh/anaconda3
     PYTHON_PATH                     :
     
     ***conda packages***
     /home/charlesh/anaconda3/bin/conda
     # packages in environment at /home/charlesh/anaconda3:
     #
     # Name                    Version                   Build  Channel
     _anaconda_depends         2020.02                  py37_0
     _ipyw_jlab_nb_ext_conf    0.1.0                    py37_0
     _libgcc_mutex             0.1                        main
     alabaster                 0.7.12                   py37_0
     anaconda                  custom                   py37_1
     anaconda-client           1.7.2                    py37_0
     anaconda-navigator        1.9.6                    py37_0
     anaconda-project          0.8.4                      py_0
     argh                      0.26.2                   py37_0
     asn1crypto                1.3.0                    py37_0
     astroid                   2.4.0                    py37_0
     astropy                   4.0.1.post1      py37h7b6447c_0
     atomicwrites              1.4.0                      py_0
     attrs                     19.3.0                     py_0
     autopep8                  1.4.4                      py_0
     babel                     2.8.0                      py_0
     backcall                  0.1.0                    py37_0
     backports                 1.0                        py_2
     backports.os              0.1.1                    py37_0
     backports.shutil_get_terminal_size 1.0.0                    py37_2
     beautifulsoup4            4.9.0                    py37_0
     bitarray                  1.2.1            py37h7b6447c_0
     bkcharts                  0.2                      py37_0
     blas                      1.0                         mkl
     blaze                     0.11.3                   py37_0
     bleach                    3.1.4                      py_0
     blosc                     1.16.3               hd408876_0
     bokeh                     2.0.2                    py37_0
     boto                      2.49.0                   py37_0
     bottleneck                1.3.2            py37heb32a55_0
     bzip2                     1.0.8                h7b6447c_0
     ca-certificates           2020.1.1                      0
     cairo                     1.14.12              h8948797_3
     certifi                   2020.4.5.1               py37_0
     cffi                      1.14.0           py37h2e261b9_0
     chardet                   3.0.4                 py37_1003
     click                     7.1.1                      py_0
     cloudpickle               1.4.1                      py_0
     clyent                    1.2.2                    py37_1
     colorama                  0.4.3                      py_0
     conda                     4.8.3                    py37_0
     conda-build               3.17.6                   py37_0
     conda-env                 2.6.0                         1
     conda-package-handling    1.6.1            py37h7b6447c_0
     conda-verify              3.1.1                    py37_0
     contextlib2               0.6.0.post1                py_0
     cryptography              2.8              py37h1ba5d50_0
     cugraph                   0.13.0a0+251.g9a87431          pypi_0    pypi
     curl                      7.69.1               hbc83047_0
     cycler                    0.10.0                   py37_0
     cython                    0.29.17          py37he6710b0_0
     cytoolz                   0.10.1           py37h7b6447c_0
     dask                      2.15.0                     py_0
     dask-core                 2.15.0                     py_0
     datashape                 0.5.4                    py37_1
     dbus                      1.13.14              hb2f20db_0
     decorator                 4.4.2                      py_0
     defusedxml                0.6.0                      py_0
     diff-match-patch          20181111                   py_0
     distributed               2.15.2                   py37_0
     docutils                  0.16                     py37_0
     entrypoints               0.3                      py37_0
     et_xmlfile                1.0.1                    py37_0
     expat                     2.2.6                he6710b0_0
     fastcache                 1.1.0            py37h7b6447c_0
     filelock                  3.0.12                     py_0
     flake8                    3.7.9                    py37_0
     flask                     1.1.2                      py_0
     flask-cors                3.0.8                      py_0
     fontconfig                2.13.0               h9420a91_0
     freetype                  2.9.1                h8a8886c_1
     fribidi                   1.0.5                h7b6447c_0
     fsspec                    0.7.1                      py_0
     future                    0.18.2                   py37_0
     get_terminal_size         1.0.0                haa9412d_0
     gevent                    1.4.0            py37h7b6447c_0
     glib                      2.63.1               h5a9c865_0
     glob2                     0.7                        py_0
     gmp                       6.1.2                h6c8ec71_1
     gmpy2                     2.0.8            py37h10f8cd9_2
     graphite2                 1.3.13               h23475e2_0
     greenlet                  0.4.15           py37h7b6447c_0
     gst-plugins-base          1.14.0               hbbd80ab_1
     gstreamer                 1.14.0               hb453b48_1
     h5py                      2.10.0           py37h7918eee_0
     harfbuzz                  1.8.8                hffaf4a1_0
     hdf5                      1.10.4               hb1b8bf9_0
     heapdict                  1.0.1                      py_0
     html5lib                  1.0.1                    py37_0
     hypothesis                5.8.3                      py_0
     icu                       58.2                 he6710b0_3
     idna                      2.9                        py_1
     imageio                   2.8.0                      py_0
     imagesize                 1.2.0                      py_0
     importlib_metadata        1.5.0                    py37_0
     intel-openmp              2020.0                      166
     intervaltree              3.0.2                      py_0
     ipykernel                 5.1.4            py37h39e3cac_0
     ipython                   7.13.0           py37h5ca1d4c_0
     ipython-autotime          0.1                      pypi_0    pypi
     ipython_genutils          0.2.0                    py37_0
     ipywidgets                7.5.1                      py_0
     isort                     4.3.21                   py37_0
     itsdangerous              1.1.0                    py37_0
     jbig                      2.1                  hdba287a_0
     jdcal                     1.4.1                      py_0
     jedi                      0.15.2                   py37_0
     jeepney                   0.4.3                      py_0
     jinja2                    2.11.2                     py_0
     joblib                    0.14.1                     py_0
     jpeg                      9b                   h024ee3a_2
     json5                     0.9.4                      py_0
     jsonschema                3.2.0                    py37_0
     jupyter                   1.0.0                    py37_7
     jupyter_client            6.1.3                      py_0
     jupyter_console           6.1.0                      py_0
     jupyter_core              4.6.3                    py37_0
     jupyterlab                1.2.6              pyhf63ae98_0
     jupyterlab_server         1.1.1                      py_0
     keyring                   21.1.1                   py37_2
     kiwisolver                1.2.0            py37hfd86e86_0
     krb5                      1.17.1               h173b8e3_0
     lazy-object-proxy         1.4.3            py37h7b6447c_0
     ld_impl_linux-64          2.33.1               h53a641e_7
     libarchive                3.3.3                h5d8350f_5
     libcurl                   7.69.1               h20c2e04_0
     libedit                   3.1.20181209         hc058e9b_0
     libffi                    3.2.1                hd88cf55_4
     libgcc-ng                 9.1.0                hdf63c60_0
     libgfortran-ng            7.3.0                hdf63c60_0
     liblief                   0.10.1               he6710b0_0
     libpng                    1.6.37               hbc83047_0
     libsodium                 1.0.16               h1bed415_0
     libspatialindex           1.9.3                he6710b0_0
     libssh2                   1.9.0                h1ba5d50_1
     libstdcxx-ng              9.1.0                hdf63c60_0
     libtiff                   4.1.0                h2733197_0
     libtool                   2.4.6                h7b6447c_5
     libuuid                   1.0.3                h1bed415_2
     libxcb                    1.13                 h1bed415_1
     libxml2                   2.9.9                hea5a465_1
     libxslt                   1.1.33               h7d1a2b0_0
     llvmlite                  0.26.0           py37hd408876_0
     locket                    0.2.0                    py37_1
     lxml                      4.5.0            py37hefd8a0e_0
     lz4-c                     1.8.1.2              h14c3975_0
     lzo                       2.10                 h7b6447c_2
     markupsafe                1.1.1            py37h7b6447c_0
     matplotlib                3.1.3                    py37_0
     matplotlib-base           3.1.3            py37hef1b27d_0
     mccabe                    0.6.1                    py37_1
     mistune                   0.8.4            py37h7b6447c_0
     mkl                       2020.0                      166
     mkl-service               2.3.0            py37he904b0f_0
     mkl_fft                   1.0.15           py37ha843d7b_0
     mkl_random                1.1.0            py37hd6b4f25_0
     mock                      4.0.2                      py_0
     more-itertools            8.2.0                      py_0
     mpc                       1.1.0                h10f8cd9_1
     mpfr                      4.0.1                hdf1c602_3
     mpmath                    1.1.0                    py37_0
     msgpack-python            1.0.0            py37hfd86e86_1
     multipledispatch          0.6.0                    py37_0
     navigator-updater         0.2.1                    py37_0
     nbconvert                 5.6.1                    py37_0
     nbformat                  5.0.4                      py_0
     ncurses                   6.2                  he6710b0_1
     networkx                  2.4                        py_0
     nltk                      3.4.5                    py37_0
     nose                      1.3.7                    py37_2
     notebook                  6.0.3                    py37_0
     numba                     0.41.0           py37h962f231_0
     numexpr                   2.7.1            py37h423224d_0
     numpy                     1.18.1           py37h4f9e942_0
     numpy-base                1.18.1           py37hde5b4d6_1
     numpydoc                  0.9.2                      py_0
     odo                       0.5.1                    py37_0
     olefile                   0.46                     py37_0
     openpyxl                  3.0.3                      py_0
     openssl                   1.1.1g               h7b6447c_0
     packaging                 20.3                       py_0
     pandas                    1.0.3            py37h0573a6f_0
     pandoc                    2.2.3.2                       0
     pandocfilters             1.4.2                    py37_1
     pango                     1.42.4               h049681c_0
     parso                     0.5.2                      py_0
     partd                     1.1.0                      py_0
     patchelf                  0.10                 he6710b0_0
     path                      13.2.0                   py37_0
     path.py                   12.0.2                     py_0
     pathlib2                  2.3.5                    py37_0
     pathtools                 0.1.2                      py_1
     patsy                     0.5.1                    py37_0
     pcre                      8.43                 he6710b0_0
     pep8                      1.7.1                    py37_0
     pexpect                   4.8.0                    py37_0
     pickleshare               0.7.5                    py37_0
     pillow                    7.1.2            py37hb39fc2d_0
     pip                       20.0.2                   py37_1
     pixman                    0.38.0               h7b6447c_0
     pkginfo                   1.5.0.1                  py37_0
     pluggy                    0.13.1                   py37_0
     ply                       3.11                     py37_0
     prometheus_client         0.7.1                      py_0
     prompt-toolkit            3.0.4                      py_0
     prompt_toolkit            3.0.4                         0
     psutil                    5.7.0            py37h7b6447c_0
     ptyprocess                0.6.0                    py37_0
     py                        1.8.1                      py_0
     py-lief                   0.10.1           py37h403a769_0
     pycodestyle               2.5.0                    py37_0
     pycosat                   0.6.3            py37h7b6447c_0
     pycparser                 2.20                       py_0
     pycrypto                  2.6.1            py37h14c3975_9
     pycurl                    7.43.0.5         py37h1ba5d50_0
     pydocstyle                4.0.1                      py_0
     pyflakes                  2.1.1                    py37_0
     pygments                  2.6.1                      py_0
     pylint                    2.5.0                    py37_1
     pyodbc                    4.0.30           py37he6710b0_0
     pyopenssl                 19.1.0                   py37_0
     pyparsing                 2.4.7                      py_0
     pyqt                      5.9.2            py37h05f1152_2
     pyrsistent                0.16.0           py37h7b6447c_0
     pysocks                   1.7.1                    py37_0
     pytables                  3.6.1            py37h71ec239_0
     pytest                    5.4.1                    py37_0
     pytest-arraydiff          0.3              py37h39e3cac_0
     pytest-astropy            0.8.0                      py_0
     pytest-astropy-header     0.1.2                      py_0
     pytest-doctestplus        0.5.0                      py_0
     pytest-openfiles          0.5.0                      py_0
     pytest-remotedata         0.3.2                    py37_0
     python                    3.7.4                h265db76_1
     python-dateutil           2.8.1                      py_0
     python-jsonrpc-server     0.3.4                      py_0
     python-language-server    0.31.10                  py37_0
     python-libarchive-c       2.9                        py_0
     pytz                      2020.1                     py_0
     pywavelets                1.1.1            py37h7b6447c_0
     pyxdg                     0.26                       py_0
     pyyaml                    5.3.1            py37h7b6447c_0
     pyzmq                     18.1.1           py37he6710b0_0
     qdarkstyle                2.8.1                      py_0
     qt                        5.9.7                h5867ecd_1
     qtawesome                 0.7.0                      py_0
     qtconsole                 4.7.3                      py_0
     qtpy                      1.9.0                      py_0
     readline                  7.0                  h7b6447c_5
     requests                  2.23.0                   py37_0
     ripgrep                   11.0.2               he32d670_0
     rope                      0.16.0                     py_0
     rtree                     0.9.4                    py37_1
     ruamel_yaml               0.15.87          py37h7b6447c_0
     scikit-image              0.16.2           py37h0573a6f_0
     scikit-learn              0.22.1           py37hd81dba3_0
     scipy                     1.4.1            py37h0b6359f_0
     seaborn                   0.10.1                     py_0
     secretstorage             3.1.2                    py37_0
     send2trash                1.5.0                    py37_0
     setuptools                46.1.3                   py37_0
     simplegeneric             0.8.1                    py37_2
     singledispatch            3.4.0.3                  py37_0
     sip                       4.19.8           py37hf484d3e_0
     six                       1.14.0                   py37_0
     snappy                    1.1.7                hbae5bb6_3
     snowballstemmer           2.0.0                      py_0
     sortedcollections         1.1.2                    py37_0
     sortedcontainers          2.1.0                    py37_0
     soupsieve                 2.0                        py_0
     sphinx                    3.0.3                      py_0
     sphinxcontrib             1.0                      py37_1
     sphinxcontrib-applehelp   1.0.2                      py_0
     sphinxcontrib-devhelp     1.0.2                      py_0
     sphinxcontrib-htmlhelp    1.0.3                      py_0
     sphinxcontrib-jsmath      1.0.1                      py_0
     sphinxcontrib-qthelp      1.0.3                      py_0
     sphinxcontrib-serializinghtml 1.1.4                      py_0
     sphinxcontrib-websupport  1.2.1                      py_0
     spyder                    4.1.2                    py37_0
     spyder-kernels            1.9.0                    py37_0
     sqlalchemy                1.3.16           py37h7b6447c_0
     sqlite                    3.31.1               h62c20be_1
     statsmodels               0.11.0           py37h7b6447c_0
     sympy                     1.5.1                    py37_0
     tbb                       2020.0               hfd86e86_0
     tblib                     1.6.0                      py_0
     terminado                 0.8.3                    py37_0
     testpath                  0.4.4                      py_0
     tk                        8.6.8                hbc83047_0
     toml                      0.10.0           py37h28b3542_0
     toolz                     0.10.0                     py_0
     tornado                   6.0.4            py37h7b6447c_1
     tqdm                      4.46.0                     py_0
     traitlets                 4.3.3                    py37_0
     typed-ast                 1.4.1            py37h7b6447c_0
     typing_extensions         3.7.4.1                  py37_0
     ujson                     1.35             py37h14c3975_0
     unicodecsv                0.14.1                   py37_0
     unixodbc                  2.3.7                h14c3975_0
     urllib3                   1.25.8                   py37_0
     watchdog                  0.10.2                   py37_0
     wcwidth                   0.1.9                      py_0
     webencodings              0.5.1                    py37_1
     werkzeug                  1.0.1                      py_0
     wheel                     0.34.2                   py37_0
     widgetsnbextension        3.5.1                    py37_0
     wrapt                     1.11.2           py37h7b6447c_0
     wurlitzer                 2.0.0                    py37_0
     xlrd                      1.2.0                    py37_0
     xlsxwriter                1.2.8                      py_0
     xlwt                      1.3.0                    py37_0
     xz                        5.2.5                h7b6447c_0
     yaml                      0.1.7                had09818_2
     yapf                      0.28.0                     py_0
     zeromq                    4.3.1                he6710b0_3
     zict                      2.0.0                      py_0
     zipp                      3.1.0                      py_0
     zlib                      1.2.11               h7b6447c_3
     zstd                      1.3.7                h0b5b093_0
     
</pre></details>
**Additional context**

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
closes #5474

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -2123,6 +2134,10 @@ def loc(self):
     def iloc(self):
         """
         Selecting rows and column by position.
+        In case result is expected to be a single
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py:2137]: This isn't always true though. It's possible to get a single row as a `DataFrame`. See Pandas behaviour (which I think we emulate, but can't confirm right now):

```python

In [4]: a.iloc[0, :]  # single row returned as a Series                 
Out[4]: 
a    1
b    2
Name: 0, dtype: int64

In [5]: a.iloc[[0,], :] # single row returned as a DF                     
Out[5]: 
   a  b
0  1  2
```

I would love for us to document the behaviour of `loc`/`iloc` w.r.t downcasting, but be warned that it's full of quirky edge cases and we don't always match Pandas behaviour. :-)

File: python/cudf/cudf/core/dataframe.py [hunk 2]
```diff
@@ -2066,6 +2066,10 @@ def iteritems(self):
     def loc(self):
         """
         Selecting rows and columns by label or boolean mask.
+        In case result is expected to be a single
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py:2069]: See below comment for `iloc`

Comment (rgsl888prabhu) [reply python/cudf/cudf/core/dataframe.py:2069]: May be I can mention the case of pandas results in a Series in case of mixed dtypes and cudf has a limtation on that.

Comment (shwina) [reply python/cudf/cudf/core/dataframe.py:2069]: Yeah - I think that would be more accurate

File: python/cudf/cudf/core/dataframe.py (hunk 1 above)
Comment (rgsl888prabhu) [reply python/cudf/cudf/core/dataframe.py:2137]: Is it for internal purpose or do you think it would be better with having it in documentation?

If it is for internal purpose, we can add test cases for all supporting edge cases.

Comment (rgsl888prabhu) [reply python/cudf/cudf/core/dataframe.py:2137]: And I can see most of them have been taken care in test cases.

File: python/cudf/cudf/core/dataframe.py [hunk 3]
```diff
@@ -2123,6 +2133,9 @@ def loc(self):
     def iloc(self):
         """
         Selecting rows and column by position.
+        If DataFrame is of mixed types and result is expected
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py:2136]: I think something like this better fits in a `Notes` section rather than the top level description. According to https://numpydoc.readthedocs.io/en/latest/format.html, the `Notes` section goes after the `See also` section.

Would suggest something like this:

```
Notes
------
One notable difference from Pandas is when returning a single row...

<example here> 
```

File: python/cudf/cudf/core/dataframe.py [hunk 4]
```diff
@@ -2184,6 +2199,21 @@ def iloc(self):
         See also
         --------
         DataFrame.loc
+
+        Notes
+        -----
+        One notable difference from Pandas is when DataFrame is of
+        mixed types and result is expected to be a Series in case of pandas,
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py:2206]:
```suggestion
        mixed types and result is expected to be a Series in case of Pandas.
```

File: python/cudf/cudf/core/dataframe.py [hunk 5]
```diff
@@ -2184,6 +2199,21 @@ def iloc(self):
...
+        cudf will return a DataFrame as it doesn't support mixed types
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py:2207]:
```suggestion
        cuDF will return a DataFrame as it doesn't support mixed types
```
